### PR TITLE
Don't unconditionally format values as metrics in generic HtmlTableView component

### PR DESCRIPTION
**Pull request by smurching**
## What changes are proposed in this pull request?

Fixes a bug where we unconditionally format values in the HtmlTableView component as metrics - the component is also used to render params, so this leads to incorrect behavior (we end up render string param values as NaN, and for string param values that have numeric values <1e-6, we throw [here](https://github.com/mlflow/mlflow/blob/19e20e42c0193e6625dd7a9801071e409c5a664d/mlflow/server/js/src/utils/Utils.js#L44))

## How is this patch tested?

Manually tested the fix. 

Before:
image

After:
image

### Is this a user-facing change?

Fixes a bug where we unconditionally format values in the HtmlTableView component as metrics - the component is also used to render params, so this leads to incorrect behavior (we end up render string param values as NaN, and for string param values that have numeric values <1e-6, we throw

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s) does this PR affect?

- [x] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@smurching Thanks for catching this error.

File: mlflow/server/js/src/components/HtmlTableView.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ class HtmlTableView extends Component {
                   return <td key={idx} style={style}>{v}</td>;
                 } else {
                   style = styles['td'];
-                  return <td key={idx} style={style} title={v}>{Utils.formatMetric(v)}</td>;
+                  return <td key={idx} style={style} title={v}>{v}</td>;
```
Comment (aarondav) [inline mlflow/server/js/src/components/HtmlTableView.js]: Hmm, doesn't this make the title no longer show the actual value?

**Comment by harupy** [comment]
What about passing a prop named `formatter` to `HtmlTableView` like below?  If that prop exists, apply that function to each value otherwise do nothing.

### RunView.js
```javascript
...
<CollapsibleSection title='Parameters'>
  <HtmlTableView
    columns={["Name", "Value"]}
    values={getParamValues(params)}
    styles={tableStyles}
  />
</CollapsibleSection>
<CollapsibleSection title='Metrics'>
  <HtmlTableView
    columns={["Name", "Value"]}
    values={getMetricValues(latestMetrics, getMetricPagePath)}
    styles={tableStyles}
    formatter={Utils.formatMetric}  // add formatter to only the metrics table
  />
</CollapsibleSection>
...
```

**Comment by smurching** [comment]
@aarondav @harupy thanks for the catch - ended up just explicitly setting the metric-value title as part of the props passed to the HtmlTableView, it seems to work :)

**Comment by harupy** [comment]
LGTM

**Pull request merged**